src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4182,9 +4189,28 @@ private bool HasTopLevelNullabilityConversion(TypeWithAnnotations source, TypeWi
             switch (conversion.Kind)
             {
                 case ConversionKind.MethodGroup:
-                    if (reportRemainingWarnings)
                     {
-                        ReportNullabilityMismatchWithTargetDelegate(location, targetType.GetDelegateType(), conversion.Method);
+                        var receiverOpt = (operandOpt as BoundMethodGroup)?.ReceiverOpt;
+                        // https://github.com/dotnet/roslyn/issues/33637: Should update method based on inferred receiver type.
+                        var method = conversion.Method;
+                        if (receiverOpt != null)
```

Comment:
This pattern is used multiple times. Consider extracting to a helper.

Reply:
Extracting helper for getting the state, but not for checking receiver since that varies slightly.

---
In reply to: [282161388](https://github.com/dotnet/roslyn/pull/35572#discussion_r282161388) [](ancestors = 282161388)